src/react-components/avatar-editor.js

```diff
@@ -0,0 +1,247 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { injectIntl, FormattedMessage } from "react-intl";
+import { getReticulumFetchUrl } from "../utils/phoenix-utils";
+import { upload } from "../utils/media-utils";
+import classNames from "classnames";
+
+import styles from "../assets/stylesheets/profile.scss";
+
+const BOT_PARENT_AVATAR = "muKnhtJ";
+
+export default class AvatarEditor extends Component {
+  static propTypes = {
+    store: PropTypes.object,
+    onSignIn: PropTypes.func,
+    onSignOut: PropTypes.func,
+    signedIn: PropTypes.bool,
+    advanced: PropTypes.bool,
+    onAvatarChanged: PropTypes.func,
+    saveStateAndFinish: PropTypes.func
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+    // this.state = { avatar: { name: "Base bot avatar", files: {} } };
+
+    this.inputFiles = {
+      glb: undefined,
+
+      base_map: undefined,
+      emissive_map: undefined,
+      normal_map: undefined,
+
+      ao_map: undefined,
+      metalic_map: undefined,
+      roughness_map: undefined,
+
+      orm_map: undefined
+    };
+  }
+
+  componentDidMount = () => {
+    if (this.props.signedIn && !this.state.avatar) {
+      return this.getPersonalAvatar();
+    }
+  };
+
+  componentDidUpdate = () => {
+    if (this.props.signedIn && !this.state.avatar) {
+      return this.getPersonalAvatar();
+    }
+  };
+
+  getPersonalAvatar = async () => {
+    const { personalAvatarId } = this.props.store.state.profile;
+    const avatar = await (personalAvatarId
+      ? this.getAvatar(personalAvatarId)
+      : this.createOrUpdateAvatar({ name: "My Avatar", parent_avatar_id: BOT_PARENT_AVATAR }));
+
+    this.props.store.update({
+      profile: {
+        personalAvatarId: avatar.avatar_id
+      }
+    });
+
+    this.props.onAvatarChanged(avatar.avatar_id);
+
+    this.setState({ ...this.state, avatar });
+  };
+
+  getAvatar = avatarId =>
+    fetch(getReticulumFetchUrl(`/api/v1/avatars/${avatarId}`), {
+      method: "GET",
```

Comment:
isn't `GET` assumed?

Reply:
it seems like a good idea to DRY this into `phoenix-utils` (which should really be called `reticulum-utils`, but I digress)